src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/RestService.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.Map;
+
+import static org.springframework.web.util.UriComponentsBuilder.fromHttpUrl;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.AUTHORIZATION_HEADER;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RestService {
+
+    private final RestTemplate restTemplate;
+
+    public void restApiPostCall(String userAuthToken, String url, Map<String, Object> body) {
```

Comment:
I think passing body as a `Map` is not the best of designs - it makes client to use `ObjectMapper` solely for the purpose of calling this method. A better design would be to use `ObjectMapper` internally here and just pass an `Object` as a body. Even better design would be to configure `RestTemplate` to use the configured `ObjectMapper`. Googling seems to confirm this is possible. Even now we're using `RestTemplate` with a plain object (see FR COS `NotificationService.java:211`) - https://docs.spring.io/spring-android/docs/current/reference/html/rest-template.html suggest a `MappingJackson2HttpMessageConverter` is going to be used and perhaps it will pick up configured `ObjectMapper`? I'm actually curious enough to write a test to see how this works. Bottom line is I would change the `body` from a `Map` to an `Object`